### PR TITLE
General simplifications

### PR DESCRIPTION
**Pull request by vitorgalvao**
The commit message isn’t great but these are a bunch of small corrections to things you’re doing the hard way.

You shouldn’t use `with input as {query}`, but `with input as argv`. With that change we shave three lines out of the script. Not only does it become simpler, it’s more robust. This is the proper way to handle multiple arguments and paths in bash and in Alfred, you don’t need to mess with IFS.

The File Buffer object is extraneous. You’re using it to process multiple files but that’s not what it’s for an it isn’t necessary for that, you can just use a File Action in multiple actions like normal.

I’ve also added an External Trigger, so people can invoke the optimisation from other Workflows. This is what prompted the PR, due to https://www.alfredforum.com/topic/17804-process-screenshots-w-imageoptim/.

I’ve only updated the files, not the Workflow file itself, because GitHub wouldn’t show the modifications to that.

If you want to add auto-updating capabilities to the Workflow, [I can help with that as well](https://www.alfredforum.com/topic/9224-oneupdater-—-update-workflows-with-a-single-node/).

### CODE REVIEW

**Comment by AlexanderWillner** [comment]
cool, thanks! will review it soon.

File: workflow/info.plist [hunk 1]
```diff
@@ -4,11 +4,9 @@
 <dict>
 	<key>bundleid</key>
 	<string>ws.willner.alfred.imageoptim</string>
-	<key>category</key>
-	<string>Tools</string>
```
Comment (AlexanderWillner) [inline workflow/info.plist:8]: why removing these lines?

Comment (vitorgalvao) [reply workflow/info.plist:8]: Nice catch, it wasn’t on purpose. I don’t use categories, so it must’ve been auto-removed. Pushing a fix now.

**Comment by benjaminhoffman** [comment]
obrigado @vitorgalvao!  i was the one who posted in the alfred forum asking about auto-processing & auto-uploading to the cloud (ie dropbox)... I'm not sure I fully follow your directions because I am new to Alfred and scripting -- do you know of a solution that's out of the box?  I was hoping to find a workflow whereby I take a screenshot and all other steps are completed for free.

**Comment by vitorgalvao** [comment]
@benjaminhoffman Please don’t post the same thing in multiple places, it splits the conversation and is worse for everyone. I’ll reply on the forum later.

**Pull request merged**